src/realm/sync/noinst/client_reset.cpp

```diff
@@ -243,10 +324,9 @@ void client_reset::transfer_group(const Transaction& group_src, Transaction& gro
 
     // Remove all columns that link to one of the tables to be removed.
     for (auto table_key : group_dst.get_table_keys()) {
```

Comment:
This is necessary before removing the table because trying to remove a table with incoming links will throw an exception "CrossTableLinkTarget". However, I see now that this check does not clean up links made by Mixed. Maybe I will have to make a version of erase_table which removes all incoming link columns to fix this.

Reply:
I changed it so that if there are any tables to remove, this will cause a client reset exception and fail. We cannot really recover from this situation anyways so might as well fail early here.